pandas/tests/reshape/concat/test_dataframe.py

```diff
@@ -167,3 +167,14 @@ def test_concat_dataframe_keys_bug(self, sort):
         # it works
         result = concat([t1, t2], axis=1, keys=["t1", "t2"], sort=sort)
         assert list(result.columns) == [("t1", "value"), ("t2", "value")]
+
+    def test_concat_duplicate_indexes(self):
+        # GH#36263 ValueError with non unique indexes
+        df1 = DataFrame([1, 2, 3, 4], index=[0, 1, 1, 4], columns=["a"])
+        df2 = DataFrame([6, 7, 8, 9], index=[0, 0, 1, 3], columns=["b"])
+        result = concat([df1, df2], axis=1)
+        expected = DataFrame(
+            {"a": [1, 1, 2, 3, np.nan, 4], "b": [6, 7, 8, 8, 9, np.nan]},
+            index=Index([0, 0, 1, 1, 3, 4]),
```

Comment:
just to make sure i understand  this, for any `df1` and `df2`, we want `result.index` to always satisfy:

```
vc = result.index.value_counts()
vc1 = df1.index.value_counts()
vc2 = df2.index.value_counts()

vc1b = vc1.reindex(vc.index, fill_value=0)
vc2b = vc2.reindex(vc.index, fill_value=0)
```

We expect `vc` to be the pointwise maximum of `vc1b` and `vc2b`?

Reply:
Yes exactly. Thats perfectly on point.